app/models/ability.rb

```diff
@@ -31,27 +31,37 @@ def initialize(user, token = nil)
       case @allowlisted_jwt.scope
       when 'all'
         can %i[create update], [Stream, Upload], organization: { allowlisted_jwts: { jti: token_payload['jti'] } }
-        can :read, [Organization, Stream, Upload]
+        can :read, Organization, public: true
+        can :read, [Stream, Upload], organization: { public: true }
+
+        can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
       when 'upload'
         can %i[create update], [Stream, Upload], organization: { allowlisted_jwts: { jti: token_payload['jti'] } }
       when 'download'
-        can %i[read], [Organization, Stream, Upload]
+        can :read, Organization, public: true
+        can :read, [Stream, Upload], organization: { public: true }
+
+        can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
       end
 
       allowlisted_jwt&.update(updated_at: Time.zone.now)
       return
     end
 
+    can :read, ActiveStorage::Attachment, { record: { organization: { public: true } } }
```

Comment:
I don't know if it's funny to write authz rules based on Rails-provided models, but 🤷‍♂️